virttest/utils_libvirt/libvirt_pcicontr.py

```diff
@@ -28,3 +28,27 @@ def get_max_contr_indexes(vm_xml, cntlr_type, cntlr_model, cntl_num=1):
                                                          cntlr_model,
                                                          usable_indexes[:cntl_num]))
     return usable_indexes[:cntl_num]
+
+
+def get_free_pci_slot(vm_xml, target_bus, max_slot=30):
+    """
+    Get a free slot for the given bus
+
+    :param vm_xml The guest xml
+    :param target_bus: The bus for the pci controller, like '0x00'
+    :param max_slot: the maximum of slot to be selected
+
+    :return: str,the first free slot or None
+    """
+    used_slot = []
+    pci_devices = vmxml.xmltreefile.find('devices').getchildren()
+    for dev in pci_devices:
+        address = dev.find('address')
+        if (address is not None and address.get('bus') == target_bus):
+            used_slot.append(address.get('slot'))
+    logging.debug("Collect used slot:%s", used_slot)
+    for slot_index in range(1, max_slot):
```

Comment:
Just one small thing: the last item of range(1, 30) is 29, is that what we want?

Reply:
@chloerh , 30 is only a default value. The user needs to change it according to his needs.

Reply:
I understand, it's just..I'm worrying the number could cause confusions, like when someone tries to get before 10, but only get the 9.
Anyway, just my opinion.

Reply:
Updated the max_slot for pcie-root controller.